detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmLockYamlParserv6.java

```diff
@@ -0,0 +1,94 @@
+package com.synopsys.integration.detectable.detectables.pnpm.lockfile.process;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.collections4.Predicate;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmLockYamlv6;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmProjectPackagev6;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.util.NameVersion;
+
+public class PnpmLockYamlParserv6 {
+    private static final Predicate<String> isNodeRoot = "."::equals;
+
+    private PnpmYamlTransformerv6 pnpmTransformer;
+
+    public PnpmLockYamlParserv6(PnpmYamlTransformerv6 pnpmTransformer) {
+        this.pnpmTransformer = pnpmTransformer;
+    }
+
+    public List<CodeLocation> parse(File parentFile, PnpmLockYamlv6 pnpmLockYaml,
+            PnpmLinkedPackageResolver linkedPackageResolver, @Nullable NameVersion projectNameVersion)
+            throws IntegrationException {
+        List<CodeLocation> codeLocationsFromImports = createCodeLocationsFromImports(parentFile, pnpmLockYaml,
+                linkedPackageResolver, projectNameVersion);
+        if (codeLocationsFromImports.isEmpty()) {
+            return createCodeLocationsFromRoot(parentFile, pnpmLockYaml, projectNameVersion, linkedPackageResolver);
+        }
+        return codeLocationsFromImports;
+    }
+
+    private List<CodeLocation> createCodeLocationsFromRoot(File sourcePath, PnpmLockYamlv6 pnpmLockYaml,
+            @Nullable NameVersion projectNameVersion, PnpmLinkedPackageResolver linkedPackageResolver)
+            throws IntegrationException {
+        CodeLocation codeLocation = pnpmTransformer.generateCodeLocation(sourcePath, pnpmLockYaml, projectNameVersion,
+                linkedPackageResolver);
+        return Collections.singletonList(codeLocation);
+    }
+
+    private List<CodeLocation> createCodeLocationsFromImports(File sourcePath, PnpmLockYamlv6 pnpmLockYaml,
```

Comment:
Would there be duplicate code locations? If yes, do we need to remove them?

Reply:
I don't think that's a concern. This is the same code as was in v5 and that area of the lockfile hasn't changed. I haven't heard of this being a concern. The imports section is an area in the lockfile that basically describes the high level dependencies of the project and then the packages section captures the entire dependency chain. I don't think you can have duplicates in the package.json, or potential npm files that you imported into pnpm, so I don't think the pnpm executable will create duplicates here.